src/content/symbol-store.js

```diff
@@ -14,70 +32,100 @@ export class SymbolStore {
    *                                which will be called whenever we need a symbol table. This method
    *                                needs to return a promise of [addr, syms] (the symbol table).
    */
-  constructor(dbNamePrefix, symbolProvider) {
+  constructor(dbNamePrefix: string, symbolProvider: SymbolProvider) {
     this._symbolProvider = symbolProvider;
-    this._db = new SymbolStoreDB(`${dbNamePrefix}-symbol-tables`);
+    this._db = new SymbolStoreDB(`${dbNamePrefix}-symbol-tables`, 200);
 
     // A set of strings identifying libraries that we have requested
     // symbols for but gotten an error back.
     this._failedRequests = new Set();
 
     // A map with one entry for each library that we have requested (but not yet
     // received) symbols for. The keys are strings (libid), and the values are
-    // promises.
-    this._importingLibs = new Map();
+    // promises that resolve to [ addrs, index, buffer ] symbol tables.
+    this._requestedSymbolTables = new Map();
+
+    // A map with one entry for each library that we have already retrieved from
+    // somewhere (either from the symbolProvider, or from this._db), and that we
+    // hold on to for a few minutes in case it's requested again.
+    this._cachedSymbolTables = new ExpirationMap(2, 2 * 60 * 1000 /* two minutes */);
   }
 
-  _ensureLibraryIsInDB(lib) {
+  _getSymbolTable(lib: Library): Promise<SymbolTableAsTuple> {
     const { debugName, breakpadId } = lib;
     const libid = `${debugName}/${breakpadId}`;
+
     if (this._failedRequests.has(libid)) {
-      return Promise.reject(new Error('We\'ve tried to request a symbol table for this library before and failed, so we\'re not trying again.'));
+      return Promise.reject(
+        new Error('We\'ve tried to request a symbol table for this library before and failed, so we\'re not trying again.'));
```

Comment:
when https://github.com/devtools-html/perf.html/pull/264/files lands you'll be able to use `common-tags`' `oneLine` function ;)